src/core/classes/CardFrames.class.ts

```
@@ -116,18 +116,25 @@ class CardFrames extends RegisterFrames {
    * @param fields
    * @param targets
    */
-  protected registerElements(fields: HTMLElement[], targets: string[]) {
-    if (fields.length && targets.length) {
+  protected registerElements(fields: HTMLElement[], targets: string[]): void {
+    if (
+      fields.length >= CardFrames.COMPLETE_FORM_NUMBER_OF_FIELDS &&
+      targets.length >= CardFrames.COMPLETE_FORM_NUMBER_OF_FIELDS &&
+      fields.length === targets.length
+    ) {
       targets.map((item, index) => {
-        document.getElementById(item).appendChild(fields[index]);
+        const element: HTMLElement = document.getElementById(item);
+        if (element) {
+          element.appendChild(fields[index]);
+        }
```

here we have some kind of protection when merchant doesn't specify fields correctly

---

maybe we should raise an error with clear message when someone wants to bind the field to the target that doesn't exist? right now it could be hard to debug why given field is not showing

---

yes, changed; I also forgot to check if the animated card is set as available or not - also fixed